lib/msf/core/framework.rb

```diff
@@ -209,6 +211,8 @@ def db
   #
   # @return [Msf::SessionManager]
   def sessions
+    return @sessions if @sessions
```

Comment:
Required to ensure that accessing framework.sessions from a trap context is safe (which can't invoke `mutex.synchronize`); it works under the assumption that the session manager is always created - which I believe is guaranteed by virtue of a session being created.

```
/Users/user/.rvm/gems/ruby-3.1.2/gems/rb-readline-0.5.5/lib/rbreadline.rb:8654: warning: Exception in finalizer #<Proc:0x0000000113636cf8 /Users/user/Documents/code/metasploit-framework/lib/rex/post/meterpreter/extensions/stdapi/sys/process.rb:340>
/Users/user/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/monitor.rb:202:in `synchronize': can't be called from trap context (ThreadError)
	from /Users/user/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/monitor.rb:202:in `mon_synchronize'
	from /Users/user/Documents/code/metasploit-framework/lib/msf/core/framework.rb:212:in `sessions'
	from /Users/user/Documents/code/metasploit-framework/lib/rex/post/meterpreter/extensions/stdapi/sys/process.rb:342:in `block in finalize'
	from /Users/user/.rvm/gems/ruby-3.1.2/gems/rb-readline-0.5.5/lib/rbreadline.rb:8654:in `[]'
	from /Users/user/.rvm/gems/ruby-3.1.2/gems/rb-readline-0.5.5/lib/rbreadline.rb:8654:in `_rl_adjust_point'
	from /Users/user/.rvm/gems/ruby-3.1.2/gems/rb-readline-0.5.5/lib/rbreadline.rb:8684:in `_rl_find_next_mbchar'
	from /Users/user/.rvm/gems/ruby-3.1.2/gems/rb-readline-0.5.5/lib/rbreadline.rb:1712:in `block in expand_prompt'
	from /Users/user/.rvm/gems/ruby-3.1.2/gems/activesupport-7.0.4.3/lib/active_support/core_ext/range/each.rb:9:in `each'
	from /Users/user/.rvm/gems/ruby-3.1.2/gems/activesupport-7.0.4.3/lib/active_support/core_ext/range/each.rb:9:in `each'
	from /Users/user/.rvm/gems/ruby-3.1.2/gems/rb-readline-0.5.5/lib/rbreadline.rb:1696:in `expand_prompt'
	from /Users/user/.rvm/gems/ruby-3.1.2/gems/rb-readline-0.5.5/lib/rbreadline.rb:1784:in `rl_expand_prompt'
	from /Users/user/.rvm/gems/ruby-3.1.2/gems/rb-readline-0.5.5/lib/rbreadline.rb:1809:in `rl_set_prompt'
	from /Users/user/.rvm/gems/ruby-3.1.2/gems/rb-readline-0.5.5/lib/rbreadline.rb:4866:in `readline'
	from /Users/user/Documents/code/metasploit-framework/lib/rex/ui/text/input/readline.rb:161:in `readline_with_output'
	from /Users/user/Documents/code/metasploit-framework/lib/rex/ui/text/input/readline.rb:99:in `pgets'
	from /Users/user/Documents/code/metasploit-framework/lib/rex/ui/text/shell.rb:329:in `get_input_line'
	from /Users/user/Documents/code/metasploit-framework/lib/rex/ui/text/shell.rb:145:in `run'
	from /Users/user/Documents/code/metasploit-framework/lib/metasploit/framework/command/console.rb:48:in `start'
	from /Users/user/Documents/code/metasploit-framework/lib/metasploit/framework/command/base.rb:82:in `start'
	from ./msfconsole:23:in `<main>'
```